src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,30 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    this.hidden = false;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(obj) {
+    if (obj instanceof Herbivore && !obj.hidden) {
+      obj.health -= 50;
+    }
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
```

Comment:
```suggestion
    if (victim,health <= 0) {
      Animal.alive = Animal.alive.filter(animal => animal.name !== victim.name)
    }
```

Reply:
if I make such edits, I fail the test.... 

![image](https://github.com/mate-academy/js_herbivores_and_carnivores/assets/155822393/43b2ebb2-a313-44ff-8efc-aa3b697e37fd)

Reply:
Oh, my bad didn't take into account that name isn't unique